flang/include/flang/Lower/AbstractConverter.h

```diff
@@ -69,6 +69,9 @@ class AbstractConverter {
   /// Get the mlir instance of a symbol.
   virtual mlir::Value getSymbolAddress(SymbolRef sym) = 0;
 
+  /// Set the mlir instance corresponding to a symbol.
+  virtual void setSymbolAddress(const SymbolRef sym, mlir::Value val) = 0;
```

Comment:
Should the variable be associated with an address? Can it be an MLIR value? I think by default if we set an argument for an MLIR block only an SSA value is created. 

Thanks @jeanPerier for the detailed comments and review.

Reply:
SSA value should be fine. If the mlir value associated to a symbol is not an address, expression lowering will not try to load it when it requires the symbol value. Of course, any Fortran code trying to assign to the symbol in the construct would lead to invalid FIR, but as I understand it, it's illegal/should be caught by semantics (you might want to check).

However, good point that it makes the the name misleading. Maybe you can drop the `Address` suffix then.

Reply:
Have added a TODO for returning a code. Did not implement now since we are using forced for till an issue is fixed.